libs/state/src/product/reducers/best-sellers.reducer.ts

```diff
@@ -0,0 +1,50 @@
+import { BestSellersActionTypes, BestSellersActions } from '../actions/best-sellers.actions';
+import { Product } from '@daffodil/core';
+
+export interface State {
+  productIds: string[],
+  loading: boolean,
+  errors: string[]
+}
+
+export const initialState: State = {
+  productIds: [],
+  loading: false,
+  errors: []
+};
+
+export const resetState: State = Object.assign({}, initialState);
+
+export function reducer(state = initialState, action: BestSellersActions): State {
+  switch (action.type) {
+    case BestSellersActionTypes.BestSellersLoadAction:
+      return {...state, loading: true};
+    case BestSellersActionTypes.BestSellersLoadSuccessAction:
+      return {...state, loading: false, productIds: getIds(action.payload)};
+    case BestSellersActionTypes.BestSellersLoadFailureAction:
+      return {...state, 
+        loading: false, 
+        errors: state.errors.concat(new Array(action.payload))
+      };
+    case BestSellersActionTypes.BestSellersResetAction:
+      return {
+        ...resetState
+      }
+    default:
+      return state;
+  }
+}
+
+export const getBestSellersIds = (state: State) => state.productIds;
+
+export const getBestSellersLoading = (state: State) => state.loading;
+
+function getIds(products: Product[]): string[] {
```

Comment:
I think this function exists on daffodil state already.

Reply:
I think the daffodil state function would return the ids of all products, because we decided to combine the state for best-sellers and regular products, right?